src/actions/TargetActions.js

```diff
@@ -17,15 +17,54 @@ export const getTargetsFail = error => ({
 
 export const getTargets = () => async dispatch => {
   try {
-    dispatch(getTargetsRequest());
+    dispatch(targetsRequest());
     const { topics } = await targetApi.fetchTopics();
     const topicsDict = topics.reduce((map, obj) => {
       map[obj.topic.id] = obj.topic;
       return map;
-    });
+    }, {});
     const { targets } = await targetApi.fetchTargets();
-    dispatch(getTargetsSuccess({ targets, topics: topicsDict }));
-  } catch (err) {
-    dispatch(getTargetsFail(err));
+    targets.forEach(target => {
+      target.target.icon = topicsDict[target.target.topicId].icon;
+    });
```

Comment:
I think this could be clearer: 
```suggestion
    const { icon, topicId } = target.target
    targets.forEach(target => {
      icon = topicsDict[topicId].icon;
    });
```

Reply:
Thanks! That code as it is doesn't work but I changed it in a way that I believe is more readable, hope you think the same way :)